src/dsp_tools/commands/xmlupload/stash/upload_stashed_resptr_props.py

```diff
@@ -31,25 +33,74 @@ def upload_stashed_resptr_props(
 
     print(f"{datetime.now()}: Upload the stashed resptrs...")
     logger.info("Upload the stashed resptrs...")
+
     upload_state.pending_stash = cast(Stash, upload_state.pending_stash)
     link_value_stash = cast(LinkValueStash, upload_state.pending_stash.link_value_stash)
-    for res_id, stash_items in link_value_stash.res_2_stash_items.copy().items():
-        res_iri = upload_state.iri_resolver.get(res_id)
+
+    _upload_process(link_value_stash=link_value_stash, iri_resolver=upload_state.iri_resolver, con=con, context=context)
+    _remove_empty_ids(link_value_stash=link_value_stash)
+
+
+def _upload_process(
+    link_value_stash: LinkValueStash,
+    iri_resolver: IriResolver,
+    con: Connection,
+    context: dict[str, str],
+) -> None:
+    """
+    Try to upload the stashed items. If the target_iri of the item can be found
+    and the item can then successfully be uploaded it is removed from link_value_stash.
+    The link_value_stash is updated accordingly, as a side effect.
+
+    Args:
+        link_value_stash: the current state of the link value stash
+        iri_resolver: service for resolving internal IDs to IRIs
+        con: connection to DSP
+        context: the JSON-LD context of the resource
+    """
+    for res_id, stash_items in link_value_stash.res_2_stash_items.items():
+        res_iri = iri_resolver.get(res_id)
         if not res_iri:
```

Comment:
I think this would be a good place for a new user warning (https://github.com/dasch-swiss/dsp-tools/blob/main/src/dsp_tools/models/custom_warnings.py).
We could tell the user, that the item could not be uploaded because a dependency failed. This may be helpful to identify follow up problems.

Reply:
I'm a bit unsure if emitting a warning is a good idea here. If a resource could not be created, the user was already alerted. And AFAIU, there will be no follow-up problems if a stashed link cannot be reapplied. 

But if you think that it would be helpful for the user, I don't want to block it.

Reply:
Well we know that a resource could not be uploaded but we do not know if that is the source of the failure to upload this stash no? I think RDU prefers us to be extremely explicit and spell out the problems. That is the reason, they often mention that they do not understand if it is a follow up problem. They don't know about the stash and how that all links.

Reply:
But wouldn't that be part of a new PR with the goal of refactoring or adding a new feature?

Reply:
> we know that a resource could not be uploaded but we do not know if that is the source of the failure to upload this stash no?

I don't agree. At this point (line 63), the only failure is that the target resource doesn't exist. The actual stash upload hasn't started yet, so there is no failure in the stash upload. 

There is a direct cause --> effect: resource could not be uploaded --> link pointing to failed resource will be skipped. There is no actual failure.

Reply:
> But wouldn't that be part of a new PR with the goal of refactoring or adding a new feature?

Yes. If we do this, then in a separate PR .